src/types/fastlane.ts

```diff
@@ -97,4 +97,55 @@ export interface IFastlaneInstance {
     };
     FastlanePaymentComponent: (options: IFastlaneComponentOptions) => Promise<IFastlanePaymentComponent>;
     FastlaneCardComponent: (options: Omit<IFastlaneComponentOptions, 'shippingAddress'>) => IFastlaneCardComponent;
+}
+
+export interface IFastlaneOptions {
+    shippingAddressOptions: IFastlaneAddressOptions,
+    cardOptions: IFastlaneCardOptions,
+    styles: IFastlaneStyleOptions
+}
+
+export interface IFastlaneAddressOptions {
+    // default: empty array = all locations allowed
+    allowedLocations: [string];
+}
+
+export interface IFastlaneCardOptions {
+    // default: empty array = all brands allowed
+    allowedBrands: [IFastlaneCardBrandEnum];
+}
+  
+export enum IFastlaneCardBrandEnum {
+    'VISA',
+    'MASTERCARD',
+    'AMERICAN-EXPRESS',
+    'DINERS-CLUB',
+    'DISCOVER',
+    'JCB',
+    'UNION-PAY',
+    'MAESTRO',
+    'ELO',
+    'MIR',
+    'HIPER',
+    'HIPERCARD',
+}
+
```

Comment:
Don't use enum. It doesn't act as a purely virtual type and actually generates code that breaks tree shaking. Use a union type instead

```suggestion

export type IFastlaneCardBrandEnum = 
  'VISA' |
  'MASTERCARD' |
  'AMERICAN-EXPRESS' |
  'DINERS-CLUB' |
  'DISCOVER' |
  'JCB' |
  'UNION-PAY' |
  'MAESTRO' |
  'ELO' |
  'MIR' |
  'HIPER' |
  'HIPERCARD';
```

Reply:
changed